roles/deploy_olm_operator/tasks/main.yml

```
@@ -1,206 +1,241 @@
+      when:
+        - csv_describe_result.stdout != ""
+        - operator_container_result.stdout != ""
+
+    - name: "Get the catalogsource pod log"
+      shell: "{{ oc_bin_path }} get --output=name pods -n openshift-marketplace | grep test-operators | cut -d' ' -f1 | xargs -I{} {{ oc_bin_path }} logs {} -n openshift-marketplace"
+      register: catalog_source_result
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+
+    - name: "Output the catalogsource log to a debug file"
+      copy:
+        content: "{{ catalog_source_result.stdout }}"
+        dest: "{{ work_dir }}/olm-catalog-source-debug.txt"
+      when: catalog_source_result.stdout is defined
+
+    - name: "Get the catalog-operator pod log"
+      shell: "{{ oc_bin_path }} get --output=name pods -n openshift-operator-lifecycle-manager | grep catalog-operator | cut -d' ' -f1 | xargs -I{} {{ oc_bin_path }} logs {} -n openshift-operator-lifecycle-manager | tail -n 100"
+      register: catalog_operator_result
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+
+    - name: "Output the catalog-operator log to a debug file"
+      copy:
+        content: "{{ catalog_operator_result.stdout }}"
+        dest: "{{ work_dir }}/olm-catalog-operator-debug.txt"
+      when: catalog_operator_result.stdout is defined
+
+    - name: "Get the InstallPlan debug output"
+      shell: "{{ oc_bin_path }} describe installplan -n {{ openshift_namespace }}"
+      register: installplan_result
+      environment:
+        KUBECONFIG: "{{ kubeconfig_path }}"
+
+    - name: "Output the InstallPlan debug output to a debug file"
+      copy:
+        content: "{{ installplan_result.stdout }}"
+        dest: "{{ work_dir }}/olm-installplan-debug.txt"
+      when: installplan_result.stdout is defined
+    
+    - name: "Set fact for deploy olm operator results"
+      set_fact:
+        deploy_olm_operator_results: {}
+
+    - name: "Failed operator deployment while waiting for operator pod deployment"
+      fail:
+        msg: 'Operator deployment with OLM failed, check the olm-*.txt files for more details'
+      when: operator_result.rc != 0
+
+  rescue:
+
+    - name: "Rescue block contains the error messages"
+      debug:
+        msg: "Rescue block has found an error, The following are details of failed task."
+
+    - name: "FAILED task name in ansible is as follows:"
+      debug:
+        msg: "{{ ansible_failed_task.name }}"
+
+    - name: "Result of failed task"
+      debug:
+        msg: "{{ ansible_failed_result }}"
+
+    - name: "Set failure result"
+      set_fact:
+        deploy_olm_operator_results: {'result': 'fail', 'name': "{{ ansible_failed_task.name }}", 'msg': "{{ ansible_failed_result }}" }
+
+    - name: "Failed on deploy_olm_operator_results"
+      fail:
+        msg: "{{ deploy_olm_operator_results }}"
+  
+  always:
+    - name: "Write parse_operator_metadata results to json file"
```

Nit: change task name to reflect the role
```suggestion
    - name: "Write deploy_olm_operator_results results to json file"
```